### ISSUE TITLE
SparkSQL: `QUALIFY` is missing as terminator for FROM keyword (also in Databricks dialect)

### ISSUE THREAD

**Issue by karabulute** [sqlfluff/sqlfluff#3929]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Using `QUALIFY` without `WHERE` clause caused the linting error by interpreting `QUALIFY` (L025) as an alias, hence generated the error below. It acts as expected with table aliasing.

> To use QUALIFY, at least one window function is required to be present in the SELECT list or the QUALIFY clause. (3)

References:

1. Previously, this has been referred in [issue#450](https://github.com/sqlfluff/sqlfluff/issues/450), but somehow closed.
2. `QUALIFY` added to `SparkSQL` in release [1.3.1](https://github.com/sqlfluff/sqlfluff/releases/tag/1.3.1) by [@ThijsKoot](https://github.com/ThijsKoot) via [#3778](https://github.com/sqlfluff/sqlfluff/pull/3778)
3. [QUALIFY clause (Databricks SQL)](https://learn.microsoft.com/en-us/azure/databricks/sql/language-manual/sql-ref-syntax-qry-select-qualify)
4. Seems [currently all tests](https://github.com/sqlfluff/sqlfluff/blob/main/test/fixtures/dialects/sparksql/select_qualify.sql) include `WHERE` clause.

### Expected Behaviour

```bash
== [C:\qualify_query.sql] PASS
==== summary ====
violations:        0 status:         PASS
All Finished!

Process finished with exit code 0
```

### Observed Behaviour

```bash
== [C:\qualify_query.sql] FAIL
L:   6 | P:   1 | L003 | Expected 1 indentation, found 0 [compared to line 05]
L:   6 | P:   1 | L011 | Implicit/explicit aliasing of table.
L:   6 | P:   1 | L014 | Unquoted identifiers must be consistently lower case.
L:   6 | P:   1 | L025 | Alias 'QUALIFY' is never used in SELECT statement.
L:   6 | P:   1 | L029 | Keywords should not be used as identifiers.
L:   6 | P:   1 | L031 | Avoid aliases in from clauses and join conditions.
==== summary ====
violations:        6 status:         FAIL
All Finished!

Process finished with exit code 1
```

### How to reproduce

```sql
SELECT
    city,
    car_model,
    RANK() OVER (PARTITION BY car_model ORDER BY quantity) AS my_rank
FROM dealer
QUALIFY my_rank <= 1
```

### Dialect

```bash
dialect:              sparksql
templater:               jinja
rules:                     all
```

### Version

```bash
sqlfluff:                1.3.2 
python:                  3.8.8
```

### Configuration

```
[sqlfluff]
dialect = databricks
templater = jinja
ignore = parsing,templating
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
